tests/test_state/test_prosumption_interface.py

```diff
@@ -0,0 +1,125 @@
+# pylint: disable=protected-access
+from unittest.mock import patch, MagicMock
+from typing import Dict
+import pytest
+from pendulum import now, DateTime
+
+from gsy_e.models.state import ProsumptionInterface
+
+
+class ProsumptionInterfaceHelper(ProsumptionInterface):
+    """Helper class to override unimplemented StateInterface's
+       abstract methods on ProsumptionInterface class."""
+
+    def get_state(self) -> Dict:
+        raise NotImplementedError
+
+    def restore_state(self, state_dict: Dict):
+        raise NotImplementedError
+
+    def delete_past_state_values(self, current_time_slot: DateTime):
+        raise NotImplementedError
+
+
+class TestProsumptionInterface:
+    """ Test ProsumptionInterface class"""
+
+    @staticmethod
+    def _setup_base_configuration():
+        return ProsumptionInterfaceHelper(), now()
+
+    def test_set_energy_measurement_kWh_dicts_not_empty(self):
+        prosumption_interface, current_time_slot = self._setup_base_configuration()
+        energy = 1
+        prosumption_interface.set_energy_measurement_kWh(
+            energy_kWh=energy, time_slot=current_time_slot
+        )
+        assert prosumption_interface.get_energy_measurement_kWh(
+            time_slot=current_time_slot) == energy
+        assert isinstance(prosumption_interface.get_forecast_measurement_deviation_kWh(
+            time_slot=current_time_slot), float)
+        assert isinstance(prosumption_interface.get_unsettled_deviation_kWh(
+            time_slot=current_time_slot), float)
+
+    @staticmethod
+    def _setup_configuration_for_settlement_posting(
+            energy_deviation=None, unsettled_deviation=None):
+        prosumption_interface = ProsumptionInterfaceHelper()
+        current_time_slot = now()
```

Comment:
You can do this here too, right?
```suggestion
   prosumption_interface, current_time_slot = self._setup_base_configuration()
```